shared/modules/doc/README.md

```diff
@@ -1,53 +1,66 @@
-# Genesis Module
+# Modules <!-- omit in toc -->
 
-This document is meant to be the development level documentation for the genesis state details related to the design of the codebase and information related to development.
+This document outlines how we structured the code splitting it into modules, what a module is and how to create one.
```

Comment:
I the first part of this sentence could be more readable by adding the word `by`.

```suggestion
This document outlines how we structured the code by splitting it into modules, what a module is and how to create one.
```